test/pointing_party/vote_calculator_test.exs

```diff
@@ -0,0 +1,87 @@
+defmodule PointingParty.VoteCalculatorTest do
+  use ExUnit.Case, async: true
+  use ExUnitProperties
+
+  describe "calculate_votes/1" do
+    setup do
+      points_map = fixed_map(%{points: integer(1..5)})
+
+      metas_map =
+        fixed_map(%{
+          metas: list_of(points_map, length: 1)
+        })
+
+      user = nonempty(map_of(string(:alphanumeric), metas_map))
+      [user: user]
+    end
+
+    property "winning value is a list or a integer", %{user: user} do
+      check all(users <- user) do
+        {_event, winner} = PointingParty.VoteCalculator.calculate_votes(users)
+        assert is_list(winner) || is_integer(winner)
+      end
+    end
+
+    property "tie when winning value is a list, winner when winning value is an integer", %{
+      user: user
+    } do
+      check all(users <- user) do
+        {event, winner} = PointingParty.VoteCalculator.calculate_votes(users)
+
+        cond do
+          is_list(winner) ->
+            assert event == "tie"
+
+          is_integer(winner) ->
+            assert event == "winner"
+        end
+      end
+    end
+
+    property "the winning value is not more than the highest point value", %{user: user} do
+      check all(users <- user) do
+        {_event, winner} = PointingParty.VoteCalculator.calculate_votes(users)
+
+        max_vote =
+          users
+          |> Enum.map(fn {_username, %{metas: [%{points: points}]}} -> points end)
+          |> Enum.max()
+
+        cond do
+          is_list(winner) ->
+            winner <= max_vote
+
+          is_integer(winner) ->
+            winner <= max_vote
+        end
+      end
+    end
+
+    property "when the winner is a list of two sorted values", %{user: user} do
+      check all(users <- user) do
+        {_event, winner} = PointingParty.VoteCalculator.calculate_votes(users)
+
+        if is_list(winner) do
```

Comment:
Since the entire set of assertions gets wrapped in a conditional, I think it would be better to set up your preconditions in such a way that `winner` will always be a list.

Reply:
So in order to create such a precondition, we have to ensure that the point value for each user shake out such that there is a tie. Given that StreamData is generating a random integer in the given range, is there anyway to control for that? I.e. to ensure that, of the random set of users, the random points for each user are each a different integer?

Reply:
The first thing you'll need to do is to change your map size so the map will always have at least two elements. One-element maps will never result in a `winner` value that is a list.

After that, there are few strategies you could employ:
* Use `filter` to discard all maps whose users all have the same point values
* Use `bind` or `bind_filter` to ensure that the second user's point value is different than the first user's point value. I'm not 100% sure how to do this, since I've never used `bind` before. I'm also not 100% sure this would work.
* Generate a one-user map. That user will have a point value in `[0, 1, 3, 5]`. Then, generate another one-user map. When you do this, your generator would remove the first user's point value from the list of possible point values for the second user. Then, you'd generate a map of zero or more users with a random point value. Finally, you'd join all the maps.

I'm fine leaving the conditional for now, since it won't be trivial to craft a generator that guarantees the results we want. I'd like to see if we can make a generator like that prior to the conference, though.